server/controllers/Conversations.js

```diff
@@ -76,63 +77,172 @@ module.exports.getConversations = async (req, res) => {
+    if (!user) return res.status(404).send({msg:"User not found"});
+    return res.send({conversationList:user["conversationList"]});
 }
 
-// Add a user to an existing conversation
-module.exports.addParticipantToConversation = async (req, res) => {
+// remove a user from a conversation
+module.exports.leaveConversation = async (req, res) => {
     let user_id = req.params.user_id;
-    //get the conversation_id and the id of the user being added to the
-    // conversation (participant_id) from the request body
+    // get the conversation_id and id of the successor admin (successor_id)) from the request body
     let conversation_id = req.body.conversation_id;
-    let participant_id = req.body.participant_id;
+    let successor_id = req.body.successor_id
     // Need to check that user_id belongs to a valid user and
     // matches the id of the requesting user
 
     // Validate request
-    if (!user_id|| !conversation_id || !participant_id) {
+    if (!user_id|| !conversation_id) {
         res.status(400).send({
-        message: "Content can not be empty!"
+        msg: "Content can not be empty!"
         });
         return;
     }
 
-    //check if the requesting user is a participant of the conversation
-    let is_participant = await Participant.findOne({
-        where: {
-            user_id: user_id,
+    // Get the participants' user ids and is_admin values, along with the number of participants
+    let participants_info = await Participant.findAndCountAll({
+        where:{
             conversation_id: conversation_id
+        },
+        attributes:["user_id", "is_admin"],
+        include:{
+            model: User,
+            as: "participantInfo",
+            attributes: ["display_name"]
         }
     })
-    if (!is_participant)
-        return res.status(404).send("requesting user is not a participant of \
-                                    conversation");
-
-    // add the new participant to the conversation using participant_id
-    // passed in the body of the request
-    let participant = await Participant.create({
-        user_id : participant_id,
-        conversation_id: conversation_id
+
+    // The number of participants
+    var participants_count = participants_info.count;
+    // The participants' user ids and is_admin values
+    var participants = participants_info.rows;
+    // the number of admins
+    var admins_count = 0;
+    // whether or not the user is a participant
+    var user_is_participant = false;
+    // whether or not the user is an admin
+    var user_is_admin = false;
+    // whether or not the successor is a participant
+    var successor_is_participant = false;
+
+    // Get the number of admins (count how many participants have is_admin set to true)
+    for (let participant of participants) {
+        if(participant.is_admin == true){
+            admins_count += 1;
+        }
+        // check if user is a participant
+        if(participant.user_id == user_id){
```

Comment:
How come the linter didn't catch this @JChris246 ? Should be 3 '=' no?

Reply:
Its editorconfig not a linter 😅

Reply:
Ok so I think we should I turn on a linter. There are some issues I see like not using triple equals and using snake case instead of camel case

Reply:
Alot of the snake case is used to maintain consistency with the database.

Reply:
Yea but it makes me sad because it very anti js. Can we fix the mapping?

Reply:
I think we'd have to rename the model properties and add aliases. Then we'd have to replace all variable usages.